jujugui/static/gui/src/app/store/env/controller-api.js

```diff
@@ -948,6 +948,11 @@ YUI.add('juju-controller-api', function(Y) {
             return prev;
           }
           prev[name] = this._parseCloudResult(result.cloud);
+          if (!prev['.names'] || !prev['.names'].length) {
```

Comment:
I guess here you could use `if (!index)` (and document this attr in the docstring), but I am not sure this is really required.
Are we doing this because we want to keep the order? But isn't the order already known by the caller, which calls this method with a list of names?